pkgs/development/python-modules/paddleocr/default.nix

```
@@ -0,0 +1,111 @@
+{ lib
+, buildPythonPackage
+, pythonRelaxDepsHook
+, fetchFromGitHub
+, attrdict
+, beautifulsoup4
+, cython
+, fire
+, fonttools
+, lmdb
+, lxml
+, numpy
+, opencv4
+, openpyxl
+, pdf2docx
+, pillow
+, premailer
+, pyclipper
+, pymupdf
+, python-docx
+, rapidfuzz
+, scikit-image
+, shapely
+, tqdm
+, paddlepaddle
+, lanms-neo
+, polygon3
+}:
+
+let
+  version = "2.7.0.1";
+in
+buildPythonPackage {
+  pname = "paddleocr";
+  inherit version;
+  format = "setuptools";
+
```

```suggestion
buildPythonPackage rec {
  pname = "paddleocr";
  version = "2.7.0.1";
  format = "setuptools";
```

This can be simplified by using a[ `rec` expression to allow self-referencing](https://nixos.wiki/wiki/Overview_of_the_Nix_Language#rec_statement).
What do you think?

---

Thank you for coming by !
As a matter of personal preference i dont like rec. My reason is that its not immediately clear which attributes are reused and which are not. Writing it like this make it obvious that only version is reused.
Many senior commiters will disagree with this, so please dont take this personally. But if possible, i would like to keep it like this.